tests/integration/servicecatalog/service_test.go

```diff
@@ -115,6 +117,45 @@ var _ = Describe("odoServiceE2e", func() {
 			helper.DeleteDir(context)
 			helper.Chdir(originalDir)
 		})
+
+		It("should be able to create, list and delete a service using a given value for --context", func() {
+			// create a component by copying the example
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "create", "nodejs", "--app", app, "--project", project)
+
+			// cd to the originalDir to create service using --context
+			helper.Chdir(originalDir)
+			helper.CmdShouldPass("odo", "service", "create", "dh-postgresql-apb", "--plan", "dev",
+				"-p", "postgresql_user=luke", "-p", "postgresql_password=secret",
+				"-p", "postgresql_database=my_data", "-p", "postgresql_version=9.6", serviceName,
+				"--context", context,
+			)
+
+			// now check if listing the service using --context works
+			ocArgs := []string{"get", "serviceinstance", "-o", "name", "-n", project}
+			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+				return strings.Contains(output, serviceName)
+			})
+			stdOut := helper.CmdShouldPass("odo", "service", "list", "--context", context)
+			Expect(stdOut).To(ContainSubstring(serviceName))
+
+			// now check if deleting the service using --context works
+			stdOut = helper.CmdShouldPass("odo", "service", "delete", "-f", "serviceName", "--context", context)
+			Expect(stdOut).To(ContainSubstring(serviceName))
+		})
+
+		It("should fail to create a service when given directory doesn't contain a context", func() {
+			// cd to the originalDir to create service using --context
+			helper.Chdir(originalDir)
+			stdOut := helper.CmdShouldFail("odo", "service", "create", "dh-postgresql-apb", "--plan", "dev",
```

Comment:
Pleas add ```SetDefaultConsistentlyDuration(30 * time.Second)``` in ```BeforeEach(func(){}``` block because ```CmdShouldFail``` has been implemented in a different way, otherwise later it will be flaky

See http://onsi.github.io/gomega/#making-asynchronous-assertions for more details.